native/libcst/src/parser/grammar.rs

```diff
@@ -0,0 +1,2930 @@
+                Expression::NamedExpr(make_named_expr(a, op, b))
+            }
+            / e:expression() !lit(":=") { e }
+
+        rule annotated_rhs() -> Expression<'a>
+            = yield_expr() / star_expressions()
+
+        rule global_stmt() -> Global<'a>
+            = kw:lit("global") init:(n:name() c:comma() {(n, c)})* last:name() {
+                make_global(kw, init, last)
+            }
+
+        rule nonlocal_stmt() -> Nonlocal<'a>
+            = kw:lit("nonlocal") init:(n:name() c:comma() {(n, c)})* last:name() {
+                make_nonlocal(kw, init, last)
+            }
+
+        rule yield_stmt() -> Expression<'a>
+            = yield_expr()
+
+        rule assert_stmt() -> Assert<'a>
+            = kw:lit("assert") test:expression() rest:(c:comma() msg:expression() {(c, msg)})? {
+                make_assert(kw, test, rest)
+            }
+
+        rule del_stmt() -> Del<'a>
+            = kw:lit("del") t:del_target() &(lit(";") / tok(NL, "NEWLINE")) {
+                make_del(kw, t)
+            }
+            / kw:lit("del") t:del_targets() &(lit(";") / tok(NL, "NEWLINE")) {
+                make_del(kw, make_del_tuple(None, t, None))
+            }
+
+        rule import_name() -> Import<'a>
+            = kw:lit("import") a:dotted_as_names() {
+                make_import(kw, a)
+            }
+
+        rule import_from() -> ImportFrom<'a>
+            = from:lit("from") dots:dots()? m:dotted_name()
+                import:lit("import") als:import_from_targets() {
+                    make_import_from(from, dots.unwrap_or_default(), Some(m), import, als)
+            }
+            / from:lit("from") dots:dots()
+                import:lit("import") als:import_from_targets() {
+                    make_import_from(from, dots, None, import, als)
+            }
+
+        rule import_from_targets() -> ParenthesizedImportNames<'a>
+            = lpar:lpar() als:import_from_as_names() c:comma()? rpar:rpar() {
+                let mut als = als;
+                if let (comma@Some(_), Some(mut last)) = (c, als.last_mut()) {
+                    last.comma = comma;
+                }
+                (Some(lpar), ImportNames::Aliases(als), Some(rpar))
+            }
+            / als:import_from_as_names() !lit(",") { (None, ImportNames::Aliases(als), None)}
+            / star:lit("*") { (None, ImportNames::Star(ImportStar {}), None) }
+
+        rule import_from_as_names() -> Vec<ImportAlias<'a>>
+            = items:separated(<import_from_as_name()>, <comma()>) {
+                make_import_from_as_names(items.0, items.1)
+            }
+
+        rule import_from_as_name() -> ImportAlias<'a>
+            = n:name() asname:(kw:lit("as") z:name() {(kw, z)})? {
+                make_import_alias(NameOrAttribute::N(n), asname)
+            }
+
+        rule dotted_as_names() -> Vec<ImportAlias<'a>>
+            = init:(d:dotted_as_name() c:comma() {d.with_comma(c)})*
```

Comment:
[nit / curiosity] another case where grammar just uses a `','.dotted_as_name+` and we don't, it seems like we could just use a comma separated rule here